tools-public/toolpad/components/DownloadsTable.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+import { createComponent } from "@mui/toolpad/browser";
+import { DataGrid } from '@mui/x-data-grid';
+import { getPackages } from "./NpmChart";
+
+export interface DownloadsTableProps {
+  data: any[]
+}
+
+const getColumns = (packages) => ([
+  { field: 'date', headerName: 'Month', width: 150 },
+  ...packages.map(packageName => ({
+    field: packageName, headerName: packageName, width: packageName === '@radix-ui/react-primitive' || packageName === '@headlessui/react' ? 170 : 120
+  })),
+  {field: 'ratio', headerName: "Base UI marketshare", width: 150}
+]);
+
+function DownloadsTable({ data: dataProp }: DownloadsTableProps) {
```

Comment:
This feels like a component that would be better in the Toolpad editor, getting us a  bit closer to the Google Sheet experience.

Reply:
I don't understand, what do you mean in the Toolpad editor?

Reply:
I'm interested in learning about why you went for a code component to implement this table rather than use the table that is built in Toolpad. Looking at the code it seems like you use the code component mainly to preprocess the incoming data.

* **Computed columns** 
    Adding columns mapped from the raw rows, but statically known. e.g. "date" and "ratio". I believe we could solve this either by providing a low-code way of remapping data, or by supporting [computed columns](https://github.com/mui/mui-toolpad/issues/137)
* **Dynamically created columns**
    Creating new columns dynamically based on incoming data. e.g. adding a column for each package. It's not clear to me yet how we can support this comprehensively, would need to benchmark other tools for ideas

Reply:
Ah I see the question, initially I was structuring the data in the component itself, but then moved this logic in a serverless function, I will try to do this change and report back if I miss a feature.

Reply:
I've updated the app to use the DataGrid component directly. The only feature I was missing (or didn't know how to do) is rename the title of the columns